plugins/woocommerce/tests/legacy/unit-tests/crud/meta.php

```diff
@@ -157,4 +157,30 @@ function test_strings_in_meta() {
 		// clean
 		$object->delete();
 	}
+
+	/**
+	 * Tests uniqueness constraint for metadata.
+	 */
+	function test_metadata_uniqueness() {
+		$object    = new WC_Order();
+		$object_id = $object->save();
+
+		// Create another reference to the same order to simulate concurrent updates.
+		$order_dup = wc_get_order( $object_id );
+
+		// Simulate two "concurrent" processes adding metadata.
+		$object->add_meta_data( 'unique_meta_key', 'a_value', true );
+		$order_dup->add_meta_data( 'unique_meta_key', 'other_value', true );
+
+		$object->save_meta_data();
+		$order_dup->save_meta_data();
+
+		// Refresh metadata.
+		$object->read_meta_data( true );
+
+		$this->assertCount( 1, $object->get_meta( 'unique_meta_key', false ) );
```

Comment:
@jorgeatorres Can we check this test fails on previous versions of WOO, just to be sure?

Reply:
NVM, I saw your earlier comment 👍 It seems like great minds think alike.

Reply:
Just to be super clear: yes, it fails on previous versions (the count returns `2` instead of `1`).